code/parachain/frame/crowdloan-rewards/src/lib.rs

```diff
@@ -307,12 +315,34 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Initialize the Crowdloan at a given block.
+		///
+		/// If the Crowdloan is overfunded by more than the `OverFundedThreshold`, the `OverFunded`
+		/// event will be emitted with the excess amount.
+		///
+		/// # Errors
+		/// * `AlreadyInitialized` - The Crowdloan has already been scheduled to start at some block
+		/// * `BackToTheFuture` - The given block, `at`, is before the current block
+		/// * `RewardsNotFunded` - The Crowdloan has not been funded with the minimum amount of
+		///   funds to provide the total rewards
 		pub(crate) fn do_initialize(at: MomentOf<T>) -> DispatchResult {
 			ensure!(!VestingTimeStart::<T>::exists(), Error::<T>::AlreadyInitialized);
+
 			let now = T::Time::now();
 			ensure!(at >= now, Error::<T>::BackToTheFuture);
+
+			let available_funds = T::RewardAsset::balance(&Self::account_id());
+			let total_rewards = TotalRewards::<T>::get();
+			ensure!(available_funds >= total_rewards, Error::<T>::RewardsNotFunded);
+			let excess_funds = available_funds - total_rewards;
```

Comment:
```suggestion
            let excess_funds = available_funds.checked_sub(total_rewards).ok_or(Error::<T>::RewardsNotFunded)?;
```

Reply:
Not sure if this is better, but it removes the `-` and the invariant required to use it safely (just makes it the same operation)

Reply:
Good catch. The actions themselves were originally separate steps, but can efficiently be taken together